### PR TITLE
Added Gatekeep feature and fixed small bugs

### PR DESCRIPTION
**Pull request by lsizemore8**
4 things happened in this PR: 
1. Beta Jar was added 
2. Fixed level 6
3. Increased default sneak value to 300 
4. Added a "gatekeep" feature to debug. If you want to test the code with the debug keys, set ALLOW_DEBUG to be true. MAKE SURE you set it back to false before merging anything, as if this is true, anyone will be able to use 'e' and 'g' to get to the debug controls while playing.

### CODE REVIEW

File: core/assets/jsons/levels/enemySurroundingPlayer.json [hunk 1]
```diff
@@ -13,6 +12,7 @@
     3.08,
     12.04
   ],
+  "bgm": "the-doll-thrown-away-in-rain",
```
Comment (BryanGraeser) [inline core/assets/jsons/levels/enemySurroundingPlayer.json:15]: Why this change?

File: core/assets/jsons/levels/enemySurroundingPlayer.json [hunk 2]
```diff
@@ -163,19 +163,19 @@
...
+        26.88,
         1.28
       ],
       "texture": "wall-side"
     },
     {
       "pos": [
         0.64,
-        11.52
+        8.96
```
Comment (BryanGraeser) [inline core/assets/jsons/levels/enemySurroundingPlayer.json:178]: Why these changes?

**Review by BryanGraeser — COMMENTED** [review]
Just a couple questions about certain changes not mentioned in the description

File: core/assets/jsons/levels/enemySurroundingPlayer.json (hunk 1 above)
Comment (lsizemore8) [reply core/assets/jsons/levels/enemySurroundingPlayer.json:15]: No actual reason. Was playing around with the level in level editor and decided it should have music.

File: core/assets/jsons/levels/enemySurroundingPlayer.json (hunk 2 above)
Comment (lsizemore8) [reply core/assets/jsons/levels/enemySurroundingPlayer.json:178]: Noticed in playtesting that there was a lot of dead space in the level. Removed it

**Review by BryanGraeser — APPROVED** [review]
gtg. Note extensive changes are bc of jar (desktop/out)
